src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1337,7 +1337,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	private _disposeOfWebglRenderer(): void {
 		this._webglAddon?.dispose();
 		this._webglAddon = undefined;
-		this._changedRendererType = false;
+		this._storageService.store(HAS_CHANGED_RENDERER_TYPE, false, StorageScope.GLOBAL, StorageTarget.USER);
```

Comment:
I think a simpler way to store would be to store the auto detected renderer, rather than storing whether the setting changed. That way we can evaluate one time which renderer seems like the best for that machine and store it in the storage service. Then all other reloads we can just query that if the key exists and the setting is `auto`. On changing the setting we can just delete the storage key.